metrics/metrics-common/src/main/java/org/apache/servicecomb/metrics/common/DoubleMetricValue.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.servicecomb.metrics.common;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.servicecomb.foundation.common.exceptions.ServiceCombException;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class DoubleMetricValue extends MetricValue<Double> {
+  public DoubleMetricValue(Double value, Map<String, String> dimensions) {
+    super(value, dimensions);
+  }
+
+  public DoubleMetricValue(@JsonProperty("key") String key,
+      @JsonProperty("value") Double value,
+      @JsonProperty("dimensions") Map<String, String> dimensions) {
+    super(key, value, dimensions);
+  }
+
+  private DoubleMetricValue merge(DoubleMetricValue value) {
+    if (this.getKey().equals(value.getKey())) {
+      return new DoubleMetricValue(this.getKey(), this.getValue() + value.getValue(), this.getDimensions());
+    }
+    throw new ServiceCombException("unable merge different key values,source key :" + value.getKey() +
```

Comment:
any place to catch this exception? we don't want to interrupt user business logic

Reply:
Done